src/components/AuthorAvatar/index.tsx

```diff
@@ -0,0 +1,41 @@
+'use client'
+
+import { Post } from '@/payload-types'
+
+import { getAuthorInitials } from '@/utilities/getAuthorInitials'
+import { format, parseISO } from 'date-fns'
+import { Avatar, AvatarFallback, AvatarImage } from '../ui/avatar'
+
+export const AuthorAvatar = (props: { authors: Post['authors']; date: Post['updatedAt'] }) => {
+  const { authors, date } = props
+
+  return (
+    <>
+      {authors?.map((author) => {
+        let initials, id, authorPhoto
+        if (author && typeof author !== 'number') {
```

Comment:
This type checking was really annoying for `author` and `date`. I am not sure if there is a better way